src/rules/noConsecutiveBlankLinesRule.ts

```diff
@@ -29,28 +29,52 @@ export class Rule extends Lint.Rules.AbstractRule {
 class NoConsecutiveBlankLinesWalker extends Lint.SkippableTokenAwareRuleWalker {
     public visitSourceFile(node: ts.SourceFile) {
         super.visitSourceFile(node);
-
-        // starting with 1 to cover the case where the file starts with two blank lines
-        let newLinesInARowSeenSoFar = 1;
-        Lint.scanAllTokens(ts.createScanner(ts.ScriptTarget.ES5, false, ts.LanguageVariant.Standard, node.text), (scanner: ts.Scanner) => {
-            const startPos = scanner.getStartPos();
-            if (this.tokensToSkipStartEndMap[startPos] != null) {
-                // tokens to skip are places where the scanner gets confused about what the token is, without the proper context
-                // (specifically, regex, identifiers, and templates). So skip those tokens.
-                scanner.setTextPos(this.tokensToSkipStartEndMap[startPos]);
-                newLinesInARowSeenSoFar = 0;
-                return;
+        
+        //Find all the lines that are blank or only contain whitespace
+        let blankLineIndexes:number[] = [];
+        node.getFullText().split(/\n/).forEach(function(txt, i){
+            if(txt.trim()===""){
+                blankLineIndexes.push(i);
             }
+        });
 
-            if (scanner.getToken() === ts.SyntaxKind.NewLineTrivia) {
-                newLinesInARowSeenSoFar += 1;
-                if (newLinesInARowSeenSoFar >= 3) {
-                    const failure = this.createFailure(scanner.getStartPos(), 1, Rule.FAILURE_STRING);
-                    this.addFailure(failure);
+        //console.log("==================")
+        //console.log("Blank or whitespace line numbers ", blankLineIndexes.map((n)=>n+1))
+        
+        //Now only keep the found blank lines that are consecutive
+        let consecutiveBlankLineStarts:number[] = [];
+        for(let i=0; i < blankLineIndexes.length; i++) {
+            let diff = blankLineIndexes[i+1] - blankLineIndexes[i];
+            if(Math.abs(diff)==1) {
+                consecutiveBlankLineStarts.push(blankLineIndexes[i]);
+            }
+        }
```

Comment:
I think you might be able to do all your logic in this one loop: keep track of something like `currentSequenceStart` and `currentSequenceLength`, and then push the value of `currentSequenceStart` into an array if the length is greater than one. Just a suggestion because the loop below feels a little complex